### PR TITLE
Added Auth and filtering for v2/apps|groups and scheduler/jobs endpoints

### PR DESCRIPTION
**Pull request by amitbose327**

### CODE REVIEW

File: aaad/aaad.py [hunk 1]
```diff
@@ -54,6 +58,14 @@ def do_GET(self):
 
         content_type = self.headers.getheader("content-type", "")
 
+        if auth_action == "filter_response":
+            filtered_response = self.filter_response(body, resource, act_as_user, permissions)
+            self.send_response(200)
```
Comment (ankanm) [inline aaad/aaad.py]: what if the response is not 200 (201 e.g.)?
what if the response is a not 2xx (say 4xx or 5xx etc.)?

File: aaad/framework.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+from __future__ import print_function
+import os
+import sys
+from abc import ABCMeta, abstractmethod
+
+class Framework(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def filterBody(self, body, allowed_namespaces, request_uri):
```
Comment (ankanm) [inline aaad/framework.py]: Consider renaming this to `filterResponseBody` to be more clear.

File: aaad/aaad.py (hunk 1 above)
Comment (amitbose327) [inline aaad/aaad.py]: When it enters the filterResponse part, it makes sure that the call flow hasn't broken earlier.

**Comment by ankanm** [comment]
LGTM

**Pull request merged**